relation_engine_server/test/test_api_v1.py

```diff
@@ -335,6 +335,18 @@ def test_fetch_invalid_stored_queries(self):
             },
         )
 
+    def test_ensure_specs(self):
```

Comment:
Just this happy path test ... I definitely need to add one for failure because that's the whole point of this. The existing file just seemed to be happy paths so I thought I'd do it 2nd pass.

Reply:
Still prototypical!

Reply:
can you add at least one error case test? I'm a bit reluctant not to have any error testing at all for this important endpoint. You can always do a simple test and then elaborate in a followup PR.

Reply:
Hehe "important"

Reply:
The error case is proving to be extra work, because mocking isn't working. Since there's no current use case for this endpoint, I propose we punt on testing the error case

Reply:
I thought that the whole point of this new endpoint was to ensure that the appropriate specs were loaded into Arango on startup, and to do something to alert the user/sys admin/whoever if they aren't -- that's your use case.

Reply:
That was my original but perhaps misguided design. The endpoint triggers the ensure specs code, but that code is now also triggered by the deployment scripts. So this endpoint actually has no use case now. It's something I didn't remove. Perhaps one day someone might want to manually change the specs in the server and check they're still valid without a re-deploy

Reply:
OK -- in that case, I think we need some documentation on release/deployment/updates of the ArangoDB and the RE API and some info on how the deployment process ensures that the specs are valid and that the appropriate analysers / indexes / etc. are in place.

I suggest creating a quick PR with just this documentation in it; we'll get Shane to review it, and then merge it in. This PR can then be rebased and you can add a bit of extra documentation to note where the spec/index/analyser checks occur.

Reply:
We should ask @scanon to take responsibility for keeping the deployment documentation up to date.

Reply:
I wrote some docs explaining that the /ensure_specs endpoint is just insurance in case you want to run it without re-deploy